ksqldb-engine/src/test/java/io/confluent/ksql/structured/SchemaKGroupedStreamTest.java

```diff
@@ -118,8 +119,10 @@ public void setUp() {
     when(keyFormatInfo.getFormat()).thenReturn(FormatFactory.KAFKA.name());
     when(keyFormat.getFormatInfo()).thenReturn(keyFormatInfo);
     when(keyFormat.withSerdeFeatures(any())).thenReturn(keyFormat);
+    when(keyFormat.getFormatInfo().copyWithoutProperty(Mockito.anyString())).thenReturn(keyFormatInfo);
```

Comment:
It'd be good to have test coverage for SCHEMA_ID removal. You can achieve that without significant changes by updating this to:
* specify the SCHEMA_ID property instead of `anyString()`
* returning a different mock format info (e.g., `keyFormatInfoWithoutSchemaId`) from the mocked call, and updating the tests to expect this new mock instead.

Reply:
I have added a unit test to validate SCHEMA_ID removal in `FormatInfoTest.java` but I don't mind adding another test here as suggested.

Reply:
The test you added to `FormatInfoTest.java` validates that SCHEMA_ID is properly removed when the helper method is called, but there was no unit test coverage that the helper method was ever called. The new unit test to added (into `SchemaKGroupedStreamTest.java` just now) validates that.

Reply:
Ah ok!